metadata-ingestion/src/datahub/ingestion/source/mode.py

```diff
@@ -1481,6 +1486,10 @@ def get_request():
                     raise HTTPError429
 
                 raise http_error
+            except JSONDecodeError as json_error:
+                raise HTTPError(
+                    f"{json_error.__class__.__name__}: {json_error}"
+                ) from json_error
```

Comment:
this seems extremely janky

let's extract out an `ModeRequestErrorTypes = (HTTPError, JSONDecodeError)`

and then use that everywhere else in the code e.g. 

```
try:
...
except ModeRequestErrorTypes as error:
...
```

Reply:
Agreed it's a bit sketch. I don't like having to change error handling in like 5 different spots though. What do you think about just catching all exceptions? Seems better in case there's a third one we're somehow missing -- in general it's never good to short circuit ingestion because of a failed request

Reply:
the idea for extracting a standard `ModeRequestErrorTypes` tuple means that we can just reuse that tuple in all our error handling, and we can easily add to the tuple in just one spot

we could also go with catching all exceptions, but it doesn't feel super clear